packages/plugin-ext/src/plugin/languages.ts

```diff
@@ -531,12 +531,18 @@ export class LanguagesExtImpl implements LanguagesExt {
     // ### Code Reference Provider end
 
     // ### Document Symbol Provider begin
-    registerDocumentSymbolProvider(selector: theia.DocumentSelector, provider: theia.DocumentSymbolProvider, pluginInfo: PluginInfo): theia.Disposable {
+    registerDocumentSymbolProvider(selector: theia.DocumentSelector, provider: theia.DocumentSymbolProvider, pluginInfo: PluginInfo,
+        metadata?: theia.DocumentSymbolProviderMetadata): theia.Disposable {
         const callId = this.addNewAdapter(new OutlineAdapter(this.documents, provider));
-        this.proxy.$registerOutlineSupport(callId, pluginInfo, this.transformDocumentSelector(selector));
+        const displayName = (metadata && metadata.label) || LanguagesExtImpl._pluginLabel(pluginInfo);
+        this.proxy.$registerOutlineSupport(callId, pluginInfo, this.transformDocumentSelector(selector), displayName);
         return this.createDisposable(callId);
     }
 
+    private static _pluginLabel(pluginInfo: PluginInfo): string {
+        return pluginInfo.displayName || pluginInfo.name;
+    }
```

Comment:
Suggestion: Having this as a `private static` method feels a bit out of place. How do you think about moving it to the other utility functions to the bottom of the file?

Reply:
Yes, definitely! Thanks for the quick review!